client/src/test/java/com/blackship/battlesheep/communication/network/AppClientSocketIT.java

```diff
@@ -1,36 +1,56 @@
 package com.blackship.battlesheep.communication.network;
 
+import com.blackship.battlesheep.common.communication.network.packet.PacketFactory;
+import com.blackship.battlesheep.common.communication.packet.Packet;
+import org.apache.commons.lang3.builder.EqualsBuilder;
 import org.testng.annotations.Test;
 
 import java.io.IOException;
 import java.net.ServerSocket;
-import java.util.Timer;
-import java.util.TimerTask;
+import java.net.Socket;
+import java.time.LocalTime;
+
+import static org.testng.Assert.assertTrue;
 
 public class AppClientSocketIT {
 
+    private static Boolean receivedPacketIsTheSameAsSent;
+
     @Test
-    public void connectAndExpectNoExceptionTest() throws IOException {
+    public void shouldConnectAndReceiveSentPackage() throws IOException, InterruptedException {
 
         //given
-        int givenPort = 8085;
-        AppClientSocket givenClientSocket = new AppClientSocket("localhost", givenPort);
+        Integer givenPort = 8085;
+        AppClientCommunicationHandler givenHandler = new AppClientCommunicationHandler(new AppClientSocket("localhost", 8085));
         ServerSocket givenServerSocket = new ServerSocket(givenPort);
+        Packet givenPacketToSend = PacketFactory.createMove().setCreationTime(LocalTime.now());
 
         //when
-        TimerTask connectTask = new TimerTask() {
-            @Override
-            public void run() {
-                try {
-                    givenClientSocket.connect();
-                } catch (IOException e) {
-                    e.printStackTrace();
-                }
+        Thread connectAndHandle = new Thread(() -> {
+            try {
+                Thread.sleep(500);
+                givenHandler.connect();
+                givenHandler.write(givenPacketToSend);
+
+                Packet receivedPacket =  givenHandler.read();
+
+                receivedPacketIsTheSameAsSent = EqualsBuilder.reflectionEquals(givenPacketToSend, receivedPacket);
+            } catch (IOException | ClassNotFoundException | InterruptedException e) {
+                e.printStackTrace();
+                receivedPacketIsTheSameAsSent = false;
             }
-        };
-        new Timer().schedule(connectTask, 500);
-        givenServerSocket.accept();
+        });
+        connectAndHandle.start();
```

Comment:
Optional: you can start() thread on the same line as you end its declaration ( ...}).start(); ), but overall 👍